test/unit/connectors/sql/test_snowflake.py

```
@@ -0,0 +1,145 @@
+from unittest.mock import MagicMock
+
+import pytest
+from pydantic import Secret
+from pytest_mock import MockerFixture
+
+from unstructured_ingest.processes.connectors.sql.snowflake import (
+    SnowflakeAccessConfig,
+    SnowflakeConnectionConfig,
+    SnowflakeUploader,
+)
+
+
+@pytest.fixture
+def snowflake_access_config():
+    return SnowflakeAccessConfig(
+        password="password",
+    )
+
+
+@pytest.fixture
+def snowflake_connection_config(snowflake_access_config: SnowflakeAccessConfig):
+    return SnowflakeConnectionConfig(
+        account="account",
+        user="user",
+        host="host",
+        port=1234,
+        database="database",
+        schema="schema",
+        role="role",
+        access_config=Secret(snowflake_access_config),
+    )
+
+
+@pytest.fixture
+def snowflake_uploader(snowflake_connection_config: SnowflakeConnectionConfig):
+    uploader = SnowflakeUploader(
+        connection_config=snowflake_connection_config,
+        upload_config=MagicMock(table_name="test_table"),
+    )
+    uploader._embeddings_dimension = None
+    return uploader
+
+
+@pytest.fixture
+def mock_cursor(mocker: MockerFixture):
+    return mocker.MagicMock()
+
+
+@pytest.fixture
+def mock_get_cursor(mocker: MockerFixture, mock_cursor: MagicMock):
+    mock = mocker.patch.context_manager(SnowflakeConnectionConfig, "get_cursor", autospec=True)
```

I think we should use:
`mocker.patch("unstructured_ingest.processes.connectors.sql.snowflake.SnowflakeConnectionConfig.get_cursor")`

So that this does not silently fail when get_cursor is not present. _Also check that path not sure if I got it right._

---

changed, thanks!